src/ivcam/sr300.cpp

```diff
@@ -69,34 +73,36 @@ namespace librealsense
         std::vector<platform::uvc_device_info> chosen;
         std::vector<std::shared_ptr<device_info>> results;
 
-        auto correct_pid = filter_by_product(uvc, { SR300_PID, SR300v2_PID });
+        auto correct_pid = filter_by_product(uvc, { SR300_PID, SR300v2_PID, SR306_PID });
         auto group_devices = group_devices_by_unique_id(correct_pid);
         for (auto& group : group_devices)
         {
-            if (group.size() == 2 &&
-                mi_present(group, 0) &&
-                mi_present(group, 2))
-            {
-                auto color = get_mi(group, 0);
-                auto depth = get_mi(group, 2);
-                platform::usb_device_info hwm;
+            platform::usb_device_info hwm;
+            platform::uvc_device_info color;
+            platform::uvc_device_info depth;
 
-                if (ivcam::try_fetch_usb_device(usb, color, hwm))
-                {
-                    auto info = std::make_shared<sr300_info>(ctx, color, depth, hwm);
-                    chosen.push_back(color);
-                    chosen.push_back(depth);
-                    results.push_back(info);
-                }
-                else
+            if (mi_present(group, 0))
+            {
+                color = get_mi(group, 0);
+                chosen.push_back(color);
+                if (!ivcam::try_fetch_usb_device(usb, color, hwm))
                 {
                     LOG_WARNING("try_fetch_usb_device(...) failed.");
+                    return results;
                 }
             }
-            else
+            if (mi_present(group, 2))
             {
-                LOG_WARNING("SR300 group_devices is empty.");
+                depth = get_mi(group, 2);
+                chosen.push_back(depth);
             }
+            if (!depth.pid) // SR306 : only mi=0 is defined
+                std::swap(color, depth);
+            auto info = std::make_shared<sr300_info>(ctx, color, depth, hwm);
+            results.push_back(info);
+
+            if (!color.pid && !depth.pid)
```

Comment:
Refactor and verify this it is consistent with the original flow + no warning for 0xaa3 SKU